api/src/main/java/org/openmrs/Cohort.java

```diff
@@ -83,7 +85,8 @@ public Cohort(String name, String description, Patient[] patients) {
 		this(name, description, (Integer[]) null);
 		if (patients != null) {
 			for (Patient p : patients) {
-				memberIds.add(p.getPatientId());
+				CohortMember cohortMember = new CohortMember(this, p, new Date());
+				members.add(cohortMember);
```

Comment:
You can combine these two lines into 1 with:

`members.add(new CohortMember(this, p, new Date()));`